src/tree-internal-api.ts

```diff
@@ -0,0 +1,48 @@
+import { TreeService } from './tree.service';
+import { Tree } from './tree';
+import { TreeModel } from './tree.types';
+import { TreeInternalComponent } from './tree-internal.component';
+import { Subject, Observable } from 'rxjs/Rx';
+import { Injectable, Inject, ElementRef } from '@angular/core';
+import * as _ from 'lodash';
+
+export class TreeInternalAPI {
+  constructor (private treeInternalComponent: TreeInternalComponent) { }
+
+  public select(e: MouseEvent = new MouseEvent('click')): void {
+    if (this.treeInternalComponent && !this.treeInternalComponent['isSelected'] && typeof this.treeInternalComponent['onNodeSelected'] === 'function') {
+      this.treeInternalComponent['onNodeSelected'](e);
+    }
+  }
+
+  public switchFoldingType(): void {
+    if (this.treeInternalComponent && typeof this.treeInternalComponent['onSwitchFoldingType'] === 'function') {
+      this.treeInternalComponent['onSwitchFoldingType']();
+    }
+  }
+
+  public expand(): void {
+    if (this.treeInternalComponent && this.treeInternalComponent['tree'] instanceof Tree) {
+      if (!this.treeInternalComponent['tree']['isNodeExpanded'].call(this.treeInternalComponent['tree'])) {
+        this.switchFoldingType();
+      }
+    }
+  }
+
+  public collapse(): void {
+    if (this.treeInternalComponent && this.treeInternalComponent['tree'] instanceof Tree) {
+      if (this.treeInternalComponent['tree']['isNodeExpanded'].call(this.treeInternalComponent['tree'])) {
```

Comment:
At some point there was a problem with this object it wan's the right one.
I remove it - the problem doesn't exists any more